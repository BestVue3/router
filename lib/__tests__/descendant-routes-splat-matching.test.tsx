import { mount } from '@vue/test-utils'
import { MemoryRouter as Router, Routes, Route, Outlet } from '..'

describe('Descendant <Routes> splat matching', () => {
    describe('when the parent route path ends with /*', () => {
        it('works', () => {
            function VueFundamentals() {
                return <h1>Vue Fundamentals</h1>
            }

            function VueCourses() {
                return (
                    <div>
                        <h1>Vue</h1>
                        <Routes>
                            <Route
                                path="vue-fundamentals"
                                element={<VueFundamentals />}
                            />
                        </Routes>
                    </div>
                )
            }

            function Courses() {
                return (
                    <div>
                        <h1>Courses</h1>
                        <Outlet />
                    </div>
                )
            }

            const wrapper = mount(() => (
                <Router initialEntries={['/courses/vue/vue-fundamentals']}>
                    <Routes>
                        <Route path="courses" element={<Courses />}>
                            <Route path="vue/*" element={<VueCourses />} />
                        </Route>
                    </Routes>
                </Router>
            ))

            expect(wrapper.html()).toMatchSnapshot()
        })
    })
})
