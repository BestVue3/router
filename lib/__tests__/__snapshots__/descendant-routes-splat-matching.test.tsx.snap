// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Descendant <Routes> splat matching when the parent route path ends with /* works 1`] = `"<div><h1>Courses</h1><div><h1>Vue</h1><h1>Vue Fundamentals</h1></div></div>"`;
